src/realm/realm_trawler.cpp

```diff
@@ -0,0 +1,572 @@
+        std::cout << "Current version: " << group.get_val(6) << std::endl;
+
+        process_free_list(group.get_val(3), group.get_val(4), group.get_val(5));
+        process_history(group.get_val(8));
+
+        auto table_names = process_names(group.get_val(0));
+
+        std::vector<DbEntry> table_entries;
+        DbEntry tables(m_is, group.get_val(1), m_refs);
+        for (unsigned i = 0; i < tables.get_size(); i++) {
+            table_entries.emplace_back(m_is, tables.get_val(i), m_refs);
+        }
+        for (unsigned i = 0; i < table_names.size(); i++) {
+            std::cout << "\nTable " << i << std::endl;
+            process_table(table_entries[i], table_names[i]);
+        }
+    }
+}
+
+std::vector<Entry> RealmFile::check_refs()
+{
+    std::vector<Entry> leaked;
+
+    unsigned start = sizeof(Header);
+    std::vector<Entry> combined(m_refs.size() + free_list.size());
+    auto it = std::set_union(m_refs.begin(), m_refs.end(), free_list.begin(), free_list.end(), combined.begin());
+    combined.resize(it - combined.begin());
+    Entry previous(0,0);
+    for (auto& a : combined) {
+        if (previous.start + previous.length > a.start) {
+            std::cerr << err_txt << "Overlapping area:" << std::endl;
+            std::cerr << err_txt << "    " << previous << std::endl;
+            std::cerr << err_txt << "    " << a << std::endl;
+        }
+        if (a.start > start) {
+            Entry e(start, a.start - start);
+            leaked.push_back(e);
+        }
+        start = a.start + a.length;
+        previous = a;
+    }
+    return leaked;
+}
+
+void RealmFile::check_leaked()
+{
+    std::vector<Entry> leaked = check_refs();
+    for (auto& a : leaked) {
+        std::cerr << err_txt << std::hex << "Leaked space: pos: 0x" << a.start << ", size: 0x" << a.length << std::dec
+                  << std::endl;
+        std::set<Entry> local_refs;
+        unsigned start = a.start;
+        unsigned end = a.start + a.length;
+        while (start < end) {
+            DbEntry db_entry(m_is, start, local_refs);
+            start += 8;
+            if (db_entry.is_valid()) {
+                start += db_entry.get_byte_size();
+            }
+        }
+
+        for (auto& r : local_refs) {
+            auto it = m_refs.find(r);
+            if (it == m_refs.end()) {
+                std::cerr << err_txt << "   Found ref: " << r << std::endl;
+            }
+        }
+    }
+}
+
+RealmFile::RealmFile(std::ifstream& is)
+    : m_is(is)
+{
+    union Buffer {
+        Header file_header;
+        char plain[sizeof(Header)];
+    } buffer;
+
+    is.seekg (0);
+    is.read(buffer.plain, sizeof(Header));
+    if (is) {
+        unsigned topref = unsigned(buffer.file_header.m_top_ref[buffer.file_header.m_flags]);
+        process_group(topref);
+        // process_group(is, file_header.m_top_ref[0]);
+        // process_group(is, file_header.m_top_ref[1]);
```

Comment:
Seems like these lines are from an older version of the tool, can they be removed?